src/main/java/reposense/authorship/model/AuthorshipSummary.java

```diff
@@ -1,36 +1,57 @@
 package reposense.authorship.model;
 
 import java.util.HashMap;
+import java.util.LinkedHashMap;
 import java.util.List;
+import java.util.Map;
 
 import reposense.model.Author;
+import reposense.model.Format;
 
 /**
  * Stores the contribution summary of the authors in the repo.
  */
 public class AuthorshipSummary {
     private final List<FileResult> fileResults;
-    private final HashMap<Author, Integer> authorFinalContributionMap;
+    private final Map<Author, Integer> authorFinalContributionMap;
+    private final Map<Author, Map<String, Integer>> authorFileFormatContributionMap;
```

Comment:
Since the `LinkedHashMap` is important in this change, better specify it in the variable declaration.
E.g.
`private final Map<Author, LinkedHashMap<String, Integer>> authorFileFormatContributionMap;`